src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -140,25 +141,24 @@ internal static bool DoSslHandshake(IntPtr sslContextPtr, IntPtr recvPtr, int re
                     error = sendCount;
                     Marshal.FreeHGlobal(sendPtr);
                     sendPtr = IntPtr.Zero;
-                    sendCount = 0;
-                    error = GetSslError(context.sslPtr, error, "Read Bio failed: ");                 
-                    throw CreateSslException("SSL Handshake failed", error);                   
+                    sendCount = 0;                  
+                    throw CreateSslException(context.sslPtr, "Read Bio failed: ", error);                   
                 }
             }
         
-            return ((libssl.SSL_state(context.sslPtr) == libssl.SslState.SSL_ST_OK)) ? true : false;
+            return ((libssl.SSL_state(context.sslPtr) == (int)libssl.SslState.SSL_ST_OK));
```

Comment:
SSL_state is an openssl API, used directly as an interop. Is the suggestion around creating a wrapper over that returns SslState enum? Currently SslState enum have only those values that we need. In case of a wrapper, we need to have all the possible values  in the enum.